### ISSUE TITLE
Contrast Ratio Calculation

### ISSUE THREAD

**Issue by jsfi** [MoOx/color#75]
Hi,
I am working with your library in a current project and need [contrast ratio calculation](http://leaverou.github.io/contrast-ratio/) between two colors.

Would you consider a pull request to add this functionality?

Thank you for your work on the library.

**Comment by Qix-** [comment]
Hi there!

Can you elaborate a bit on exactly what you mean?

**Comment by jsfi** [comment]
Sorry if this was a bit short.

I believe that the calculation of the contrast ratio between two colors as defined by the [W3C](http://www.w3.org/TR/2008/REC-WCAG20-20081211/#contrast-ratiodef) would be a good addition to the library.

Lea Verou implemented this [calculation](https://github.com/LeaVerou/contrast-ratio/blob/gh-pages/color.js#L108) for a [Web-Tool](http://leaverou.github.io/contrast-ratio/) and I adapted it to use with your library.

So my question was if you would be interested that I add my adaption inside the library or do you consider it out of scope.

**Comment by Qix-** [comment]
I would consider it out of scope simply because it's a destructive calculation. For instance, you can derive a ratio between two colors, but a single color cannot result in a contrast ratio and vice versa a contrast ratio cannot be used to create a specific color, hence landing this outside the scope of this library.

Unless, that is, I'm misunderstanding the concept here (which is entirely possible).

**Comment by jsfi** [comment]
What you described was my main concern why I wanted to ask before doing anything.

Thank you for your time.

**Comment by Qix-** [comment]
No problem! Thank you for the issue :) :+1:
